lib/ex_aws/sns.ex

```diff
@@ -169,6 +169,77 @@ defmodule ExAws.SNS do
     request("ListSubscriptions", %{"NextToken" => next_token})
   end
 
+  ## Messages
+  ######################
+
+  @notification_params ["Message", "MessageId", "Subject", "Timestamp", "TopicArn", "Type"]
+  @optional_notification_params ["Subject"]
+  @confirmation_params ["Message", "MessageId", "SubscribeURL", "Timestamp", "Token", "TopicArn", "Type"]
+  @signature_params ["SignatureVersion", "Signature", "SigningCertURL"]
+  @message_types ["SubscriptionConfirmation", "UnsubscribeConfirmation", "Notification"]
+
+  @doc "Verify message signature"
+  def verify_message(message_params) do
+    with :ok <- validate_message_params(message_params),
+      :ok <- validate_signature_version(message_params["SignatureVersion"]),
+      {:ok, public_key} <- ExAws.SNS.PublicKeyCache.get(message_params["SigningCertURL"]) do
+        message_params
+        |> get_string_to_sign
+        |> verify(message_params["Signature"], public_key)
+    else
+      {:error, error} -> {false, error}
+    end
+  end
+
+  defp validate_message_params(message_params) do
+    with {:ok, required_params} <- get_required_params(message_params["Type"]) do
+      case required_params -- Map.keys(message_params) do
+        [] -> :ok
+        missing_params -> {:error, "The following parameters are missing: #{inspect missing_params}"}
+      end        
+    end
+  end
+
+  defp get_required_params(message_type) do
+    case message_type do
+      "Notification" -> {:ok, (@notification_params -- @optional_notification_params) ++ @signature_params}
+      "SubscriptionConfirmation" -> {:ok, @confirmation_params ++ @signature_params}
+      "UnsubscribeConfirmation" -> {:ok, @confirmation_params ++ @signature_params}
```

Comment:
Those types are indeed coming from Amazon SNS documentation. It should always be one of those.